2025-09-29 23:12:37.821425: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-09-29 23:12:38.438355: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 14618 MB memory:  -> device: 0, name: Quadro RTX 5000, pci bus id: 0000:3b:00.0, compute capability: 7.5
2025-09-29 23:12:39.039289: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)

=== A_lr_0.001_drop_0.0_l2_0.0 ===
arch=[256, 128] lr=0.001 dropout=0.0 l2=0.0 epochs=10
Epoch 1/10
1688/1688 - 8s - loss: 0.8163 - accuracy: 0.7037 - val_loss: 0.5645 - val_accuracy: 0.7878
Epoch 2/10
1688/1688 - 5s - loss: 0.5350 - accuracy: 0.8065 - val_loss: 0.4875 - val_accuracy: 0.8203
Epoch 3/10
1688/1688 - 5s - loss: 0.4702 - accuracy: 0.8303 - val_loss: 0.4478 - val_accuracy: 0.8387
Epoch 4/10
1688/1688 - 5s - loss: 0.4340 - accuracy: 0.8428 - val_loss: 0.4166 - val_accuracy: 0.8492
Epoch 5/10
1688/1688 - 5s - loss: 0.4087 - accuracy: 0.8536 - val_loss: 0.4042 - val_accuracy: 0.8528
Epoch 6/10
1688/1688 - 5s - loss: 0.3907 - accuracy: 0.8577 - val_loss: 0.3924 - val_accuracy: 0.8577
Epoch 7/10
1688/1688 - 5s - loss: 0.3757 - accuracy: 0.8643 - val_loss: 0.3866 - val_accuracy: 0.8615
Epoch 8/10
1688/1688 - 5s - loss: 0.3626 - accuracy: 0.8689 - val_loss: 0.3884 - val_accuracy: 0.8577
Epoch 9/10
1688/1688 - 5s - loss: 0.3531 - accuracy: 0.8709 - val_loss: 0.3677 - val_accuracy: 0.8663
Epoch 10/10
1688/1688 - 5s - loss: 0.3424 - accuracy: 0.8759 - val_loss: 0.3583 - val_accuracy: 0.8677
VAL  -> loss=0.3583  acc=0.8677
TEST -> loss=0.3923 acc=0.8588
Traceback (most recent call last):
  File "/home/cse479/msaltzman/CSCE479-Homeworks/HW1/main.py", line 83, in <module>
    main()
  File "/home/cse479/msaltzman/CSCE479-Homeworks/HW1/main.py", line 64, in main
    run(name, hidden_units, learning_rate, reg['dropout'], reg['l2'], train_ds, val_ds, test_ds)
  File "/home/cse479/msaltzman/CSCE479-Homeworks/HW1/main.py", line 26, in run
    test_conf_matrix = util.confusion_matrix(model, test_ds).numpy()
  File "/home/cse479/msaltzman/CSCE479-Homeworks/HW1/util.py", line 48, in confusion_matrix
    p = tf.argmax(model(x, training=False), axis=-1)
NameError: name 'tf' is not defined
